lua/cmake-tools/executors/terminal.lua

```diff
@@ -542,6 +542,20 @@ function terminal.run(cmd, env, args, cwd, opts)
   -- Reposition the terminal buffer, before sending commands
   local final_win_id = terminal.reposition(opts)
 
+  --- NOTE: env_script needs to be run only once if the terminal buffer does not already exist
+  --- We compare the old and the new id and only if they are not the same, plus if the terminal exists,
+  --    only then, we do not reinitialize the environment, else we reinit the env
+  if terminal_already_exists or terminal.id_old ~= terminal.id then
+    terminal.id_old = terminal.id
+    terminal.send_data_to_terminal(buffer_idx, env_script, {
+      win_id = final_win_id,
+      split_direction = opts.split_direction,
+      split_size = opts.split_size,
+      start_insert = opts.start_insert_in_other_tasks,
+      focus_on_main_terminal = opts.focus_on_main_terminal,
+    })
+  end
+
```

Comment:
Why should we run the script if the terminal already exists? I think we should only run the script when the terminal did not exist or terminal.id_old not equals terminal.id.

Reply:
Ah... I had an error for this, when I used `:bclose` or `:bdelete` on the terminal buffer and vim threw an error for this, saying buffer not found. That is why I made it something like this...

> This should ensure env_script is exists
>>  if not terminal_already_exists or terminal.id_old ~= terminal.id then

However, I was not able to reproduce the error, so I noted it down and added extra checking: `if not terminal_already_exists or ... `

Reply:
Update: 

Nope, if the  "not" is not present, the env script runs each time, so we need:

```bash
if not terminal_already_exists or terminal.id_old ~= terminal.id then
  ...
end
```